adr/0007-use-rust-binary-for-both-injection-stages.md

```diff
@@ -0,0 +1,30 @@
+# 7. Use Rust Binary for Both Injection Stages
+
+Date: 2022-10-29
+
+## Status
+
+Accepted
+
+Amends [3. Image injection into remote clusters without native support](0003-image-injection-into-remote-clusters-without-native-support.md)
+
+## Context
+
+In ADR 3, the decision was made to create a rust binary (`stage1`) that would re-assemble a `registry:2` image and a go registry binary (`stage2`) from a series of configmaps. While this solution works, it is overkill for the operations that `stage2` performs. The `stage2` binary is only responsible for 1. starting a docker registry in `rw` mode, 2. pushing the `registry:2` crane tarball into said registry, 3. starting the docker registry in `r` mode. This `registry:2` image is then immedietely consumed by the `zarf-registry` package, creating a true in-cluster docker registry. The injector pod is then destroyed. The overhead this operation creates:
+
+- having to keep track of another binary (making the total number 3 for the zarf ecosystem)
+- nearly doubling the amount of configmaps loaded into the cluster (makes init slower)
+- having to compile the binary for each platform (adds to the build time + ci)
+- using a full-featured docker registry to host a single image (wasteful)
+
+## Decision
+
+The rust binary is already being injected via configmap and unpacking the tarball. There is little need to bring everthing but the kitchen sink to just serve a single image. Therefore the decision is to use the rust binary to perform the entire injection process. This required a few changes to the rust binary:
+
+- not only re-assemble + unpack the tarball, but also unpack the `registry:2` image (stored as a [crane tarball format](https://github.com/google/go-containerregistry/tree/main/pkg/v1/tarball))
+- transform the `registry:2` crane manifest to a docker v2 manifest
+- spin up a HTTP server compliant with the v2 docker registry API to serve the `registry:2` image
```

Comment:
```suggestion
- spin up an HTTP server compliant with the v2 docker registry API to serve the `registry:2` image
```
When an acronym is pronounced with a vowel in front you use an: https://www.grammar.com/a-vs-an-when-to-use/